src/journal/sd-journal.c

```diff
@@ -1661,6 +1661,9 @@ static int add_search_paths(sd_journal *j) {
         NULSTR_FOREACH(p, search_paths)
                 (void) add_root_directory(j, p, true);
 
+        if (!j->flags & SD_JOURNAL_LOCAL_ONLY)
```

Comment:
this doesn't look right. you are using a boolean negation of the flags field. I figure you mean:

```c
if (!(j->flags & SD_JOURNAL_LOCAL_ONLY))
```

Other than that I think the patch looks OK